lib/sla.js

```diff
@@ -1,6 +1,93 @@
 'use strict';
+var Q = require('q');
 
-module.exports = function(callback){
-    console.log("\"snarflegluff\" should be printed below:");
-    var str = "snarflegluff";
-    return str;}
+/**
+ * Looks for an existing mixing bridge and creates one if none exist.
+ * @param {Object} client - Object that contains information from the ARI
+ *   connection.
+ * @return {Q} Q promise object.
+ */
+function designateMixingBridge(client) {
+  var defer = Q.defer();
+  var list = Q.denodeify(client.bridges.list.bind(client.bridges));
+  list().then(
+      function(bridges) {
+        var bridge = bridges.filter(function(candidate) {
+          return candidate['bridge_type'] === 'mixing';
+        })[0];
+        if (bridge) {
+          console.log('Using existing mixing bridge %s', bridge.id);
+          defer.resolve(bridge);
+        } else {
+          client.bridges.create({type: 'mixing'}, function(err, bridge) {
+            console.log('Created new mixing bridge %s', bridge.id);
+            defer.resolve(bridge);
+          });
+        }
+      })
+    .catch(function (err) {
+      defer.reject(err);
+    });
+  return defer.promise;
+}
+
+/**
+ * Adds created channel to the bridge.
+ * @param {Object} channel - Created channel to be added to the bridge.
+ * @param {Object} bridge - Bridge that channel is to be added to.
+ * @return {Q} Q promise object.
+ */
+function addChannelsToBridge(channel, bridge) {
+  var defer = Q.defer();
+  console.log('Adding channel to bridge %s', bridge.id);
+  var addChannel = Q.denodeify(bridge.addChannel.bind(bridge));
+  addChannel({channel: channel.id})
+    .catch(function (err) {
+      defer.reject(err);
+    });
+  bridge.once('ChannelEnteredBridge', function(event, bridge) {
+    console.log('Channel %s has entered the bridge', bridge.channel.id);
+  });
+  bridge.once('ChannelLeftBridge', function(event, bridge) {
+    console.log('Channel %s has left the bridge', bridge.channel.id);
+    defer.resolve('Application completed');
```

Comment:
That's fine for this first step, but I don't think you want to ultimately have the promise resolve only once the channel hangs up. Events are a better way of informing users of your API that something like that has happened.